src/cpp/src/image_generation/models/unet_inference_static_bs1.hpp

```diff
@@ -0,0 +1,144 @@
+        OPENVINO_ASSERT(m_nativeBatchSize && m_nativeBatchSize == m_requests.size(),
+                        "UNet model must be compiled first");
+
+        size_t encoder_hidden_states_bs = encoder_hidden_states.get_shape()[0];
+        if (encoder_hidden_states_bs != m_nativeBatchSize)
+        {
+            throw std::runtime_error("UNetInferenceStaticBS1::set_hidden_states: native batch size is " + std::to_string(m_nativeBatchSize) 
+                + ", but encoder_hidden_states has batch size of " + std::to_string(encoder_hidden_states_bs));
+        }
+
+        char* pHiddenStates = (char *)encoder_hidden_states.data();
+        size_t hidden_states_batch_stride_bytes = encoder_hidden_states.get_strides()[0];
+
+        for (int i = 0; i < m_nativeBatchSize; i++)
+        {
+            auto hidden_states_bs1 = m_requests[i].get_tensor(tensor_name);
+
+            // wrap current pHiddenStates location as batch-1 tensor.
+            ov::Tensor bs1_wrapper(hidden_states_bs1.get_element_type(),
+                                   hidden_states_bs1.get_shape(),
+                                   pHiddenStates,
+                                   encoder_hidden_states.get_strides());
+
+            // copy it to infer request batch-1 tensor
+            bs1_wrapper.copy_to(hidden_states_bs1);
+
+            // increment pHiddenStates to start location of next batch (using stride)
+            pHiddenStates += hidden_states_batch_stride_bytes;
+        }
+    }
+
+    virtual void set_adapters(AdapterController& adapter_controller, const AdapterConfig& adapters) override {
+        OPENVINO_ASSERT(m_nativeBatchSize && m_nativeBatchSize == m_requests.size(),
+                        "UNet model must be compiled first");
+        for (int i = 0; i < m_nativeBatchSize; i++) {
+            adapter_controller.apply(m_requests[i], adapters);
+        }
+    }
+
+    virtual ov::Tensor infer(ov::Tensor sample, ov::Tensor timestep) override {
+        OPENVINO_ASSERT(m_nativeBatchSize && m_nativeBatchSize == m_requests.size(),
+                        "UNet model must be compiled first");
+
+        char* pSample = (char *)sample.data();
+        size_t sample_batch_stride_bytes = sample.get_strides()[0];
+
+        for (int i = 0; i < m_nativeBatchSize; i++) {
+            m_requests[i].set_tensor("timestep", timestep);
+
+            auto sample_bs1 = m_requests[i].get_tensor("sample");
+
+            // wrap current pSample location as batch-1 tensor.
+            ov::Tensor bs1_wrapper(sample_bs1.get_element_type(), sample_bs1.get_shape(), pSample, sample.get_strides());
+
+            // copy it to infer request batch-1 tensor
+            bs1_wrapper.copy_to(sample_bs1);
+
+            //increment pSample to start location of next batch (using stride)
+            pSample += sample_batch_stride_bytes;
+
+            // kick off infer for this request.
+            m_requests[i].start_async();
```

Comment:
will all these requests are executed concurrently on NPU? 
In OpenVINO for CPU / GPU you typically need to set throughput stream > 1 to ensure concurrency

Reply:
I don't think they actually would be executed concurrently. The main intent with using async here was so some 'copy_to' logic could potentially overlap with inference, at least if the native batch size is > 1.

Reply:
addressed in this PR: https://github.com/openvinotoolkit/openvino.genai/pull/1172

Reply:
but, in a theory, can they be executed concurrently on NPU?